scripts/setup.sh

```diff
@@ -179,18 +177,59 @@ get_ext_files()
   done
 }
 
+get_extensions_zip_packages() {
+  curl -s -O $EXTENSION_S3_URL/$EXTENSION_MANIFEST_FILE
+  mkdir -p ./extensions
+
+  grep -v '^ *#' < "$EXTENSION_MANIFEST_FILE" | while IFS= read -r EXTENSION_FILE_NAME
+  do 
+    (cd ./extensions && curl -s -O "$EXTENSION_S3_URL/$EXTENSION_FILE_NAME")
+  done
+}
+
 check_if_parameter_is_empty()
 {
   PARAMETER=$1
   PARAMETER_NAME=$2
-  if [ "$PARAMETER" == "null" ] || [ -z "$PARAMETER" ]
-  then
+  if [ "$PARAMETER" == "null" ] || [ -z "$PARAMETER" ]; then
     warn "Missing required parameter: $PARAMETER_NAME. Please set proper value in ./activation-config.yaml or delete it to fetch latest version automatically"
     exit
   fi
 }
 
+check_api_token() {
+  DYNATRACE_URL=$1
+  DYNATRACE_ACCESS_KEY=$2
+  V1_API_REQUIREMENTS=("ReadConfig" "WriteConfig")
+  V2_API_REQUIREMENTS=("extensions.read" "extensions.write" "extensionConfigurations.read" "extensionConfigurations.write" "extensionEnvironment.read" "extensionEnvironment.write")
+  
+  if RESPONSE=$(curl -k -s -X POST -d "{\"token\":\"$DYNATRACE_ACCESS_KEY\"}" "$DYNATRACE_URL/api/v2/apiTokens/lookup" -w "<<HTTP_CODE>>%{http_code}" -H "accept: application/json; charset=utf-8" -H "Content-Type: application/json; charset=utf-8" -H "Authorization: Api-Token $DYNATRACE_ACCESS_KEY" --connect-timeout 20); then
+    CODE=$(sed -rn 's/.*<<HTTP_CODE>>(.*)$/\1/p' <<<"$RESPONSE")
+    RESPONSE=$(sed -r 's/(.*)<<HTTP_CODE>>.*$/\1/' <<<"$RESPONSE")
+    
+    if [ "$CODE" -ge 300 ]; then
+      echo -e "\e[91mERROR: \e[37mFailed to check Dynatrace API token permissions - please verify provided values for parameters: --target-url (${DYNATRACE_URL}) and --target-api-token. $RESPONSE"
+      exit 1
+    fi
 
+    for REQUIRED in "${V1_API_REQUIREMENTS[@]}"; do
+      if ! grep -q "$REQUIRED" <<<"$RESPONSE"; then
+        echo -e "\e[91mERROR: \e[37mMissing $REQUIRED permission (v1) for the API token"
+        exit 1
+      fi
+    done
+
+    for REQUIRED in "${V2_API_REQUIREMENTS[@]}"; do
+      if ! grep -q "$REQUIRED" <<<"$RESPONSE"; then
+        echo -e "\e[91mERROR: \e[37mMissing $REQUIRED permission (v2) for the API token"
+        exit 1
+      fi
+    done
+
+  else
+      echo -e "\e[93mWARNING: \e[37mFailed to connect to endpoint $DYNATRACE_URL to check API token permissions. It can be ignored if Dynatrace/ActiveGate does not allow public access."
```

Comment:
Please remove `ActiveGate` - deployment with cloud function supports only metrics (AG can be installed only with helm for log forwarder)